ethereum/core/src/main/java/tech/pegasys/teku/core/stategenerator/StateRegenerationBaseSelector.java

```diff
@@ -121,13 +121,20 @@ private boolean isBetterThanCurrentRebasedStartingPoint(
             .isGreaterThan(closestAvailableFromStore.getSlot())) {
       return SafeFuture.completedFuture(rebasedStartingPoint);
     }
+
     return blockProvider
         .getBlock(closestAvailableFromStore.getBlockRoot())
         .thenApply(
-            maybeBlock ->
-                maybeBlock.map(
+            maybeBlock -> {
+              if (maybeBlock.isPresent()) {
+                return maybeBlock.map(
                     block ->
-                        StateAndBlockSummary.create(block, closestAvailableFromStore.getState())));
+                        StateAndBlockSummary.create(block, closestAvailableFromStore.getState()));
+              } else {
+                return Optional.of(
+                    StateAndBlockSummary.create(closestAvailableFromStore.getState()));
+              }
+            });
```

Comment:
If we don't actually need the block, would it be better to always create the summary purely off of the state and avoid looking up via blockProvider entirely?  I worry that if the block is sometimes available and sometimes not, it creates more corner cases that could be hard to test.  Whereas if it's never present we know the downstream code has to be able to handle it.

Reply:
Good point - I'll follow up on this suggestion in another PR.  I wanted to avoid changing the current production behavior in this PR, but it does make sense to simplify where we can.

Reply:
Added a follow-up ticket here to make sure I don't forget: https://github.com/ConsenSys/teku/issues/3184